delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -544,6 +551,87 @@ final class Files(
     }
   }
 
+  override def migrate(
+      id: Iri,
+      projectRef: ProjectRef,
+      rev: Option[Long],
+      storage: StorageReference,
+      attributes: kg.FileAttributes
+  )(implicit caller: Subject): IO[MigrationRejection, Unit] =
+    for {
+      project               <- projects.fetchActiveProject(projectRef).mapError(MigrationRejection(_))
+      (storageRef, storage) <- storages
+                                 .fetch(Revision(storage.id, storage.rev), project.ref)
+                                 .map(Revision(storage.id, storage.rev) -> _.value)
+                                 .leftWiden[FileRejection]
+                                 .mapError(MigrationRejection(_))
+      digest                 = if (attributes.digest == kg.Digest.empty) NotComputedDigest
+                               else
+                                 ComputedDigest(
+                                   DigestAlgorithm(attributes.digest.algorithm).getOrElse(DigestAlgorithm.default),
+                                   attributes.digest.value
+                                 )
+      origin                 = digest match {
+                                 case NotComputedDigest => FileAttributesOrigin.Storage
+                                 case _                 => FileAttributesOrigin.Client
+                               }
+      attributes15           = FileAttributes(
+                                 attributes.uuid,
+                                 attributes.location,
+                                 attributes.path,
+                                 attributes.filename,
+                                 attributes.mediaType,
+                                 attributes.bytes,
+                                 digest,
+                                 origin
+                               )
+      res                   <- eval(
+                                 MigrateFile(id, projectRef, storageRef, storage.tpe, attributes15, rev.getOrElse(0L), caller),
+                                 project
+                               ).void.mapError(MigrationRejection(_))
+    } yield res
+
+  override def fileAttributesUpdated(id: Iri, projectRef: ProjectRef, rev: Long, attributes: StorageFileAttributes)(
+      implicit subject: Subject
+  ): IO[MigrationRejection, Unit] = {
+    val digest =
+      if (attributes.digest == kg.Digest.empty) NotComputedDigest
```

Comment:
You seem to have this logic in more than one place. Why not refactoring in a `private def` ?